src/py42/modules/archive.py

```diff
@@ -7,6 +7,114 @@ def __init__(self, archive_accessor_manager, archive_client):
+
+    def get_by_user_uid(self, user_uid):
+        """Gets archive information for a user.
+        `REST Documentation <https://console.us.code42.com/apidocviewer/#Archive-get>`__
+
+        Args:
+            user_uid (int): The userUid for a user.
+
+        Returns:
+            generator: An object that iterates over :class:`py42.response.Py42Response`
+            objects that each contain a page of archives.
+        """
+        return self._archive_client.get_archives_from_value(user_uid, u"userUid")
+
+    def get_by_store_point_id(self, store_point_id):
+        """Gets archive information for a store point.
+        `REST Documentation <https://console.us.code42.com/apidocviewer/#Archive-get>`__
+
+        Args:
+            store_point_id (int): The storePointId for a storepoint.
+
+        Returns:
+            generator: An object that iterates over :class:`py42.response.Py42Response`
+            objects that each contain a page of archives.
+        """
+        return self._archive_client.get_archives_from_value(store_point_id,
+                                                            u"storePointId")
+
+    def get_by_server_guid(self, server_guid):
+        """Gets archive information for a server.
+        `REST Documentation <https://console.us.code42.com/apidocviewer/#Archive-get>`__
+
+        Args:
+            server_guid (int): The serverGuid for a server.
+
+        Returns:
+            generator: An object that iterates over :class:`py42.response.Py42Response`
+            objects that each contain a page of archives.
+        """
+        return self._archive_client.get_archives_from_value(server_guid, u"serverGuid")
+
+    def get_by_destination_guid(self, destination_guid):
+        """Gets archive information for a destination.
+        `REST Documentation <https://console.us.code42.com/apidocviewer/#Archive-get>`__
+
+        Args:
+            destination_guid (int): The destinationGuid for a destination.
+
+        Returns:
+            generator: An object that iterates over :class:`py42.response.Py42Response`
+            objects that each contain a page of archives.
+        """
+        return self._archive_client.get_archives_from_value(destination_guid,
+                                                            u"destinationGuid")
+
+    def get_by_plan_uid(self, plan_uid):
+        """Gets archive information for a destination.
+        `REST Documentation <https://console.us.code42.com/apidocviewer/#Archive-get>`__
+
+        Args:
+            plan_uid (int): The planUid for a device.
+
+        Returns:
+            generator: An object that iterates over :class:`py42.response.Py42Response`
+            objects that each contain a page of archives.
+        """
+        return self._archive_client.get_archives_from_value(plan_uid,
+                                                            u"planUid")
+
+    def get_by_uid_list(self, user_uid_list):
```

Comment:
`get_by_user_uids()` ?

Reply:
I don't really like that there is a separate method expecting a list.  We should just check the if it not a list using the `isinstance` builtin method, and if it is not a list, that wrapping it between `[]`

```
if not isinstance(user_uids, (list, tuple)):
    user_uids = [user_uids]
```

Reply:
See my note in the main comment section, I would actually prefer that this method not exist at all, though i'm of course open to listening if theres a particularly strong use case for it.

Reply:
This method has been removed.